meteor/client/ui/SegmentTimeline/SegmentTimeline.tsx

```diff
@@ -668,17 +672,71 @@ export class SegmentTimelineClass extends React.Component<Translated<IProps>, IS
 							onPieceDoubleClick={this.props.onItemDoubleClick}
 							scrollWidth={this.state.timelineWidth / this.props.timeScale}
 							firstPartInSegment={this.props.parts[0]}
+							lastPartInSegment={this.props.parts[this.props.parts.length - 1]}
 							isLastSegment={this.props.isLastSegment}
-							isLastInSegment={index === this.props.parts.length - 1}
+							isLastInSegment={index === this.props.parts.length - 1 && this.props.budgetGap <= 0}
 							isAfterLastValidInSegmentAndItsLive={
 								index === (this.props.lastValidPartIndex || 0) + 1 &&
 								previousPartIsLive &&
 								!!this.props.playlist.nextPartInstanceId
 							}
 							part={part}
+							isBudgetGap={false}
 						/>
 					)
 				})}
+				{this.props.budgetGap > 0 && (
+					<SegmentTimelinePart
+						key={'gap'}
```

Comment:
Only arrays of elements need a `key` property in React. It's not needed otherwise.